node/external/transactionAPI/apiTransactionResults.go

```diff
@@ -0,0 +1,222 @@
+	addressPubKeyConverter core.PubkeyConverter
+	historyRepository      dblookupext.HistoryRepository
+	storageService         dataRetriever.StorageService
+	marshalizer            marshal.Marshalizer
+	selfShardID            uint32
+}
+
+func newAPITransactionResultProcessor(
+	addressPubKeyConverter core.PubkeyConverter,
+	historyRepository dblookupext.HistoryRepository,
+	storageService dataRetriever.StorageService,
+	marshalizer marshal.Marshalizer,
+	selfShardID uint32,
+) *apiTransactionResultsProcessor {
+	return &apiTransactionResultsProcessor{
+		addressPubKeyConverter: addressPubKeyConverter,
+		historyRepository:      historyRepository,
+		storageService:         storageService,
+		marshalizer:            marshalizer,
+		selfShardID:            selfShardID,
+	}
+}
+
+func (arp *apiTransactionResultsProcessor) putResultsInTransaction(hash []byte, tx *transaction.ApiTransactionResult, epoch uint32) {
+	arp.putLogsInTransaction(hash, tx, epoch)
+
+	resultsHashes, err := arp.historyRepository.GetResultsHashesByTxHash(hash, epoch)
+	if err != nil {
+		return
+	}
+
+	if len(resultsHashes.ReceiptsHash) > 0 {
+		arp.putReceiptInTransaction(tx, resultsHashes.ReceiptsHash, epoch)
+		return
+	}
+
+	arp.putSmartContractResultsInTransaction(tx, resultsHashes.ScResultsHashesAndEpoch)
+}
+
+func (arp *apiTransactionResultsProcessor) putReceiptInTransaction(tx *transaction.ApiTransactionResult, recHash []byte, epoch uint32) {
+	rec, err := arp.getReceiptFromStorage(recHash, epoch)
+	if err != nil {
+		log.Warn("nodeTransactionEvents.putReceiptInTransaction() cannot get receipt from storage",
+			"hash", hex.EncodeToString(recHash))
+		return
+	}
+
+	tx.Receipt = arp.adaptReceipt(rec)
+}
+
+func (arp *apiTransactionResultsProcessor) getReceiptFromStorage(hash []byte, epoch uint32) (*receipt.Receipt, error) {
+	receiptsStorer := arp.storageService.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	receiptBytes, err := receiptsStorer.GetFromEpoch(hash, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	rec := &receipt.Receipt{}
+	err = arp.marshalizer.Unmarshal(rec, receiptBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return rec, nil
+}
+
+func (arp *apiTransactionResultsProcessor) adaptReceipt(rcpt *receipt.Receipt) *transaction.ApiReceipt {
+	return &transaction.ApiReceipt{
+		Value:   rcpt.Value,
+		SndAddr: arp.addressPubKeyConverter.Encode(rcpt.SndAddr),
+		Data:    string(rcpt.Data),
+		TxHash:  hex.EncodeToString(rcpt.TxHash),
+	}
+}
+
+func (arp *apiTransactionResultsProcessor) putSmartContractResultsInTransaction(
+	tx *transaction.ApiTransactionResult,
+	scrHashesEpoch []*dblookupext.ScResultsHashesAndEpoch,
+) {
+	for _, scrHashesE := range scrHashesEpoch {
+		for _, scrHash := range scrHashesE.ScResultsHashes {
+			scr, err := arp.getScrFromStorage(scrHash, scrHashesE.Epoch)
+			if err != nil {
+				log.Warn("putSmartContractResultsInTransaction cannot get result from storage",
+					"hash", hex.EncodeToString(scrHash),
```

Comment:
no need for `hex.EncodeToString(scrHash)` as the logger will automatically display it right. can be `scrHash`

Reply:
done.